projects/sandbox/src/app/wizard/wizard.component.ts

```diff
@@ -96,18 +96,37 @@ export class WizardComponent extends DataPollingComponent {
   }
 
   /**
-   * validatorWhitespace
+   * validateDatasetId
    *
-   * form validator implementation for whitespace detection
+   * form validator implementation for dataset id field
    *
    * @param { FormControl } control - the control to validate
    * @returns null or a code-keyed boolean
    **/
-  validatorWhitespace(control: FormControl): { [key: string]: boolean } | null {
+  validateDatasetId(control: FormControl): { [key: string]: boolean } | null {
     const val = control.value;
+    const sVal = `${val}`;
     if (val) {
-      const isWhitespace = val.length > 0 && val.trim().length === 0;
-      if (isWhitespace) {
+      if (isNaN(val) || sVal.length === 0 || sVal.length !== `${parseInt(val)}`.length) {
+        return { invalid: true };
+      }
+    }
+    return null;
+  }
```

Comment:
This seems overly complicated and also doesn't do exactly what we want, I think? What about `-4`? Or perhaps weird scientific notation (`5e4` or something like that)?

In my mind it's best to do it like below in the name validator. Use a regex to make sure that it matches only digits.

Reply:
Fixed